src/main/java/ch/njol/skript/registrations/Classes.java

```diff
@@ -113,7 +113,7 @@ public static void onRegistrationsStop() {
 				final ClassInfo<?> sa = getExactClassInfo(ci.getSerializeAs());
 				if (sa == null) {
 					Skript.error(ci.getCodeName() + "'s 'serializeAs' class is not registered");
-				} else if (sa.getSerializer() == null) {
+				} else if (sa.getSerializer() == null && ci.getSerializeAs() != Object.class) {

```

Comment:
do you think it would be a good idea to only evaluate this second part if persistent data exists? it may help limit impact if this ends up causing a different issue

Reply:
I don’t think it will matter really. No other ClassInfos should really be using Object here unless they encounter this specific issue. Maybe I’m misunderstanding what you mean though?